test/e2e/scheduler_predicates.go

```diff
@@ -900,16 +886,37 @@ func getRequestedCPU(pod v1.Pod) int64 {
 	return result
 }
 
-func waitForScheduler() {
-	// Wait a bit to allow scheduler to do its thing
-	// TODO: this is brittle; there's no guarantee the scheduler will have run in 10 seconds.
-	framework.Logf("Sleeping 10 seconds and crossing our fingers that scheduler will run in that time.")
-	time.Sleep(10 * time.Second)
+func waitForScheduler(f *framework.Framework, PodName string, expectSuccess bool) {
+	framework.Logf("Waiting for scheduler...")
+	action := func() error { // No action before waiting for Scheduler.
+		return nil
+	}
+	predicate := scheduleFailure(PodName)
+	if expectSuccess {
+		predicate = scheduleSuccess(PodName, "" /* any node */)
+	}
+	success, err := observeEventAfterAction(f, predicate, action)
+	Expect(err).NotTo(HaveOccurred())
+	Expect(success).To(Equal(true))
+}
+
+func createPausePodAndWaitForScheduler(f *framework.Framework, conf pausePodConfig, expectSuccess bool) {
+	action := func() error {
+		_, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).Create(initPausePod(f, conf))
+		return err
+	}
+	predicate := scheduleFailure(conf.Name)
```

Comment:
We should probably pull `scheduleFailure` and `scheduleSuccess` predicate factories out of `opaque_resource.go`.